### PR TITLE
Add Ubuntu 23.10 (Mantic Minotaur) RIDs to .NET 7

### PR DESCRIPTION
**Pull request by dviererbe**
main PR #85912 

# Description

Adds the Ubuntu 23.10 (Mantic Minotaur) RIDs to .NET 7

* library Microsoft.NETCore.Platforms
   - added RIDs to `runtime.compatibility.json`
   - added RIDs to `runtime.json`
   - added Mantic Version Id to Ubuntu `runtimeGroups.props`
   - incremented `ServicingVersion` in `Microsoft.NETCore.Platforms.csproj`

# Customer Impact

Will be able to target their build specifically for Ubuntu 23.10 (Mantic Minotaur) or include Ubuntu 23.10 (Mantic Minotaur) platform-specific assets.

# Regression

No.

# Testing

1. From a clean copy of your branch, run `build.sh libs -allconfigurations`
2. Check in `artifacts/bin/packages/Debug` for the existence of `Microsoft.NETCore.Platforms`, with the appropriate package version. 
    - `test -d artifacts/bin/Microsoft.NETCore.Platforms/ && echo "Ok"`
    - `test -e artifacts/bin/Microsoft.NETCore.Platforms/Debug/net7.0/Microsoft.NETCore.Platforms.BuildTasks.dll && echo "Ok"`
3. Try installing the built package in a test application, testing that your changes to the library are present & working as expected.
    - `grep ubuntu.23.10 artifacts/bin/Microsoft.NETCore.Platforms/Debug/runtime.json.dgml` shows 
      ```
      <Node Id="ubuntu.23.10" />
      <Node Id="ubuntu.23.10-x64" />
      <Node Id="ubuntu.23.10-x86" />
      <Node Id="ubuntu.23.10-arm" />
      <Node Id="ubuntu.23.10-arm64" />
      <Link Source="ubuntu.23.10" Target="ubuntu" />
      <Link Source="ubuntu.23.10-x64" Target="ubuntu.23.10" />
      <Link Source="ubuntu.23.10-x64" Target="ubuntu-x64" />
      <Link Source="ubuntu.23.10-x86" Target="ubuntu.23.10" />
      <Link Source="ubuntu.23.10-x86" Target="ubuntu-x86" />
      <Link Source="ubuntu.23.10-arm" Target="ubuntu.23.10" />
      <Link Source="ubuntu.23.10-arm" Target="ubuntu-arm" />
      <Link Source="ubuntu.23.10-arm64" Target="ubuntu.23.10" />
      <Link Source="ubuntu.23.10-arm64" Target="ubuntu-arm64" />
      ```
    - Check if `Microsoft.NETCore.Platforms` Library functions are callable
        ```bash
        export DOTNET_REPO=$PWD
        mkdir ../TestApp
        cd ../TestApp
        
        cat <<'EOF' >TestApp.csproj
        <Project Sdk="Microsoft.NET.Sdk">
            <PropertyGroup>
                <OutputType>Exe</OutputType>
                <TargetFramework>net7.0</TargetFramework>
                <ImplicitUsings>enable</ImplicitUsings>
                <Nullable>enable</Nullable>
            </PropertyGroup>
            <ItemGroup>
                <Reference Include="Microsoft.NETCore.Platforms.BuildTasks">
                    <HintPath>$(DOTNET_REPO)/artifacts/bin/Microsoft.NETCore.Platforms/Debug/net7.0/Microsoft.NETCore.Platforms.BuildTasks.dll</HintPath>
                </Reference>
            </ItemGroup>
        </Project>
        EOF
        
        cat <<EOF >Program.cs
        using Microsoft.NETCore.Platforms.BuildTasks;
        
        // Test that library functions can be called
        var rid = RID.Parse("ubuntu.23.10-x64", noQualifier: true);
        Console.WriteLine("Mantic Minotaur: " + rid.ToString());
        
        Console.WriteLine("Ok");
        EOF
        
        dotnet run
        ```

As far as I can tell the change works as expected.

Note: Builds & local Tests pass in `ppa:dviererbe/dotnet7-0-105-mantic-2` ([link](https://launchpad.net/~dviererbe/+archive/ubuntu/dotnet7-0-105-mantic-2)) with this patch applied.

# Risk

As for every change an unintended regression could be introduced. The risk for this is minimal, as the PR into main (#84983) and a recent similar PR for Ubuntu 22.10 (Kinetic Kudu) for.NET 7 (#84984) and .NET 6 (#85912) passed

# Package authoring signed off?

Adding the Ubuntu 23.10 (Mantic Minotaur) RIDs was also suggested by @wfurt (see: https://github.com/dotnet/runtime/pull/84984#issuecomment-1533445478)

### CODE REVIEW

**Comment by dviererbe** [comment]
I do not see how the "Libraries Build windows allConfigurations x64 Debug" failures 
 - `Unable to find a stable package Microsoft.Extensions.Logging.Abstractions with version (>= 7.0.1)`
 - `Unable to find a stable package System.DirectoryServices.Protocols with version (>= 7.0.1)`
 - `Unable to find a stable package System.Management with version (>= 7.0.2)`
 
 should be caused by my change

**Review by carlossanlop — CHANGES_REQUESTED** [review]
We need this in the June release so the servicing version number needs to only be bumped once (stay in 3).

File: src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <_generateRuntimeGraphTask>$([MSBuild]::NormalizePath('$(BaseOutputPath)', $(Configuration), '$(_generateRuntimeGraphTargetFramework)', '$(AssemblyName).dll'))</_generateRuntimeGraphTask>
     <!-- When building from source, ensure the RID we're building for is part of the RID graph -->
     <AdditionalRuntimeIdentifiers Condition="'$(DotNetBuildFromSource)' == 'true'">$(AdditionalRuntimeIdentifiers);$(OutputRID)</AdditionalRuntimeIdentifiers>
-    <ServicingVersion>3</ServicingVersion>
+    <ServicingVersion>4</ServicingVersion>
```
Comment (carlossanlop) [inline src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj:27]:
```suggestion
    <ServicingVersion>3</ServicingVersion>
```

**Review by wfurt — APPROVED** [review]
LGTM

**Review by wfurt — APPROVED** [review]
LGTM

**Review by carlossanlop — APPROVED** [review]

**Pull request merged**